x-pack/plugins/lens/public/datatable_visualization/components/dimension_editor.tsx

```diff
@@ -7,55 +7,121 @@
 
 import React from 'react';
 import { i18n } from '@kbn/i18n';
-import { EuiSwitch, EuiFormRow } from '@elastic/eui';
+import { EuiFormRow, EuiSwitch, EuiButtonGroup, htmlIdGenerator } from '@elastic/eui';
 import { VisualizationDimensionEditorProps } from '../../types';
 import { DatatableVisualizationState } from '../visualization';
 
+const idPrefix = htmlIdGenerator()();
+
 export function TableDimensionEditor(
   props: VisualizationDimensionEditorProps<DatatableVisualizationState>
 ) {
-  const { state, setState, accessor } = props;
-  const column = state.columns.find((c) => c.columnId === accessor);
+  const { state, setState, frame, accessor } = props;
+  const column = state.columns.find(({ columnId }) => accessor === columnId);
 
-  const visibleColumnsCount = state.columns.filter((c) => !c.hidden).length;
+  if (!column) return null;
 
-  if (!column) {
-    return null;
-  }
+  // either read config state or use same logic as chart itself
+  const currentAlignment =
+    column?.alignment ||
+    (frame.activeData &&
+    frame.activeData[state.layerId].columns.find((col) => col.id === accessor)?.meta.type ===
+      'number'
+      ? 'right'
+      : 'left');
```

Comment:
Are we sure that we should be applying right alignment to numbers without guaranteed decimal places? For example, when looking at `Average of bytes` I think the defaults got worse:

<img width="763" alt="Screen Shot 2021-02-09 at 5 27 40 PM" src="https://user-images.githubusercontent.com/666475/107436897-29c4d380-6afc-11eb-9acb-d31093fa8563.png">

Reply:
I don't think the default alignment should depend on whether or not decimal places are shown. I checked Google sheets and the Numbers app on Mac and both of them right align numbers without decimal places by default.